spec/mailers/jobapps_mailer_spec.rb

```diff
@@ -145,6 +145,39 @@
     end
   end
 
+  describe 'send_note_for_later' do
+    let(:record) do
+      create :application_record,
+             saved_for_later: true,
+             note_for_later: 'We need you to grow up a little'
+    end
+    let!(:template) do
```

Comment:
Sounds good - then you can either integrate this into your `before :each` or reference it directly as `template` in the test rather than indirectly through other objects.